utilities/transactions/transaction_util.cc

```diff
@@ -100,8 +100,8 @@ Status TransactionUtil::CheckKey(DBImpl* db_impl, SuperVersion* sv,
     SequenceNumber seq = kMaxSequenceNumber;
     bool found_record_for_key = false;
 
-    Status s = db_impl->GetLatestSequenceForKey(sv, key, !need_to_read_sst,
-                                                &seq, &found_record_for_key);
+    Status s = db_impl->GetLatestSequenceForKey(
+        sv, key, !need_to_read_sst, snap_seq, &seq, &found_record_for_key);
```

Comment:
After #4961 is landed, we should pass min_uncommitted as the lower_bound_seq so that the optimization would work for WritePrepared as well.